rollouts/controllers/rollout_controller.go

```diff
@@ -510,11 +510,18 @@ func (r *RolloutReconciler) rolloutTargets(ctx context.Context, rollout *gitopsv
 
 	for _, target := range targets.ToBeCreated {
 		rootSyncSpec := toRootSyncSpec(target.packageRef)
+		var metadata *gitopsv1alpha1.Metadata
+
+		if rollout != nil && rollout.Spec.SyncTemplate != nil &&
```

Comment:
nit: `rollout != nil` check should be moved at the beginning of the function.

One more thing I noticed since we do not support `RepoSync` yet, can we throw an error in the reconcileRollout function (I think we have some other validation bits as well).

One more ask: can we also create an issue to support `RepoSync` and Would suggest that to pick that after this.

Reply:
Filed https://github.com/GoogleContainerTools/kpt/issues/3838

Reply:
Also moved the check, and added a check to `reconcileRollout` for `RepoSync`.